plugins/scripting/scripting-compiler-impl/src/org/jetbrains/kotlin/scripting/resolve/refineCompilationConfiguration.kt

```diff
@@ -354,27 +361,56 @@ fun getScriptCollectedData(
                 jvmGetScriptingClass(ann, contextClassLoader, hostConfiguration) as? KClass<Annotation> // TODO errors
             }
         }.orEmpty()
-    val annotations = scriptFile.annotationEntries.construct(contextClassLoader, acceptedAnnotations, project)
+    val annotations = scriptFile.annotationEntries.construct(
+        contextClassLoader,
+        acceptedAnnotations,
+        project,
+        DocumentImpl(scriptFile.text),
```

Comment:
I just realized that it could be quite an expensive way to extract coordinates. Sorry for not noticing it before. Could you, please, investigate it deeper, maybe it is actually a preferable and inexpensive way, or maybe there are cheaper ways to get it from the PSI.

Reply:
The PSI only contains the reference to the file and the offsets without position information which makes sense. So we do need to convert the offset into line and column numbers.

I looked it up. I notice that this API is basically just using the LineSet API from IntelliJ, so I decided to move to using it directly, to avoid any of the additional overhead from the DocumentImpl class.

I also looked at the implementation of LineSet and I don't think we're can get any better performance than that tbh. 
I remember writing this exact same logic for my thesis: https://github.com/nerdsupremacist/Graphaello/blob/master/Sources/Graphaello/Processing/Pipeline/Component/1.%20Extraction/SourceCode/LineColumnIndex.swift